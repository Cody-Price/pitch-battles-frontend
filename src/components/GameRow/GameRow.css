.game-perfect-mark.true {
  background: url("../TeacherStudentView/images/checkmark.png");
  background-size: cover;
  height: 24px;
  width: 24px;
  margin: auto;
}
.game-perfect-mark.false {
  background: url("../TeacherStudentView/images/x-symbol.png");
  background-size: cover;
  height: 24px;
  width: 24px;
  margin: auto;
}

.game-id-code {
  width: 19px;
  margin-left: 1.1rem;
  margin-right: 2.1rem;
}

.game-row {
  display: flex;
  margin: auto;
  width: 900px;
  justify-content: space-around;
}
.game-score,
.game-score-label {
  width: 70px;
  text-align: center;
}
