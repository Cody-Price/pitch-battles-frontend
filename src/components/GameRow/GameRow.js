import React, { Component } from "react";
import "./GameRow.css";

import "../../utilities/gameHelper";

class GameRow extends Component {
  render() {
    const { game } = this.props;
    return (
      <article className="game-row">
        <p className="game-id-code">{game.id}</p>
        <p className="game-score">
          {game.attributes.level_one_duration / 1000 || "none"}
        </p>
        <div
          className={`game-perfect-mark ${game.attributes.level_one_perfect}`}
        />
        <p className="game-score">
          {game.attributes.level_two_duration / 1000 || "none"}
        </p>
        <div
          className={`game-perfect-mark ${game.attributes.level_two_perfect}`}
        />
        <p className="game-score">
          {game.attributes.level_three_duration / 1000 || "none"}
        </p>
        <div
          className={`game-perfect-mark ${game.attributes.level_three_perfect}`}
        />
        <p className="game-score">
          {game.attributes.level_four_duration / 1000 || "none"}
        </p>
        <div
          className={`game-perfect-mark ${game.attributes.level_four_perfect}`}
        />
      </article>
    );
  }
}

export default GameRow;
