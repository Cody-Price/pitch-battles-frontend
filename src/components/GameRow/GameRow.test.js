import React from "react";
import ReactDOM from "react-dom";
import GameRow from "./GameRow";
import Enzyme, { shallow } from "enzyme";
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

describe("GameRow", () => {
  let wrapper;
  const mockGame = {
    id: 3,
    attributes: {
      level_one_perfect: true,
      level_two_duration: 123,
      level_one_duration: 123,
      level_two_perfect: true,
      level_three_duration: 123,
      level_three_perfect: true,
      level_four_duration: true,
      level_four_perfect: false
    }
  };

  beforeEach(() => {
    wrapper = shallow(<GameRow game={mockGame} />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
