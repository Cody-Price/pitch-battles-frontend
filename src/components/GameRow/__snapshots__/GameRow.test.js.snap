// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameRow should match the snapshot 1`] = `
<article
  className="game-row"
>
  <p
    className="game-id-code"
  >
    3
  </p>
  <p
    className="game-score"
  >
    0.123
  </p>
  <div
    className="game-perfect-mark true"
  />
  <p
    className="game-score"
  >
    0.123
  </p>
  <div
    className="game-perfect-mark true"
  />
  <p
    className="game-score"
  >
    0.123
  </p>
  <div
    className="game-perfect-mark true"
  />
  <p
    className="game-score"
  >
    0.001
  </p>
  <div
    className="game-perfect-mark false"
  />
</article>
`;
